YourEasyRent/Controllers/ListingController.cs

```diff
@@ -38,25 +39,37 @@ public class ListingController : ControllerBase
         public async Task<IEnumerable<Product>> GetProducts()// публичный метод  GetProducts с аргументами(Section- тип аргумента, section- название аргумента)  тип  возвращаеммого значение Task<IEnumerable<Product>> 
         {
             var section = Section.Makeup;
-
-            var pagenumber = 1;
-            var sephoralistings = await _sclient.FetchFromSephoraSection(section, pagenumber); //  сделать 2 listinga для сефоры и для дугласа, смерджить и результаты(list или join  погуглить как) и вернуть общий результат для обоих сайтов 
-
-             // сделать цикл с перебором страниц, чтобы "перелистывать страницы" и брать новые ответы
-
-            var douglaslistings = await _dclient.FetchFromDouglasSection(section, pagenumber);
-            var allListings = sephoralistings.Concat(douglaslistings); 
-            // Объединяет две последовательности, сохраняя все элементы исходных последовательностей, включая дубликаты.
-            await _repository.CreateMany(allListings);
-            return allListings;
-
+            var allListings = new List<Product>();
+            for (var pagenumber = 1; pagenumber < 4; pagenumber++)
+            {
+                var sephoralistings = await _sclient.FetchFromSephoraSection(section, pagenumber); //  сделать 2 listinga для сефоры и для дугласа, смерджить и результаты(list или join  погуглить как) и вернуть общий результат для обоих сайтов 
```

Comment:
Check the implementation in my repo. In short you should have something like this

```
var clients = new List<IClient> {_sclient, _dclient};
clients.Select(x=> x.FetchFromSection(section, pagenumber).SelectMany(x=>x);
```

All your clients should implement the same interface, I'll also show you how to register them in such a way, so it is automatically put in List<IClient> .